Cargo.toml

```diff
@@ -37,6 +37,7 @@ pin-project = "1.0.11"
 rand = "0.8"
 rand_core = "0.6"
 rand_distr = "0.4.3"
+rhs_first_assign = "0.1.0"
```

Comment:
eh, I guess I was wrong claiming it is possible to use `AddAssign` in the iteration. Borrow checker would prevent that. I don't think it is worth a new dependency to be able to write `+=` instead of `+`. Shall we revert back to it?

Reply:
we can also use an intermediate variable to addassign, i will do that instead